quantum/action_tapping.c

```diff
@@ -125,29 +194,14 @@ void action_tapping_process(keyrecord_t record) {
 /* return true when key event is processed or consumed. */
 bool process_tapping(keyrecord_t *keyp) {
     keyevent_t event = keyp->event;
-#    if (defined(AUTO_SHIFT_ENABLE) && defined(RETRO_SHIFT)) || defined(TAPPING_TERM_PER_KEY) || defined(PERMISSIVE_HOLD_PER_KEY) || defined(TAPPING_FORCE_HOLD_PER_KEY) || defined(HOLD_ON_OTHER_KEY_PRESS_PER_KEY)
-    uint16_t tapping_keycode = get_record_keycode(&tapping_key, false);
-#    endif
+    DEFINE_tapping_keycode;
 
     // if tapping
     if (IS_TAPPING_PRESSED()) {
-        // clang-format off
-        if (WITHIN_TAPPING_TERM(event)
-#    if defined(AUTO_SHIFT_ENABLE) && defined(RETRO_SHIFT)
-            || (
-#        ifdef RETRO_TAPPING_PER_KEY
-                get_retro_tapping(tapping_keycode, &tapping_key) &&
-#        endif
-                (RETRO_SHIFT + 0) != 0 && TIMER_DIFF_16(event.time, tapping_key.event.time) < (RETRO_SHIFT + 0)
-            )
-#    endif
-        ) {
-            // clang-format on
+        if (WITHIN_TAPPING_TERM(event) || IS_RETRO_SHIFTING(event)) {
             if (tapping_key.tap.count == 0) {
                 if (IS_TAPPING_RECORD(keyp) && !event.pressed) {
-#    if defined(AUTO_SHIFT_ENABLE) && defined(RETRO_SHIFT)
-                    retroshift_swap_times();
-#    endif
+                    RETROSHIFT_SWAP_TIMES();
```

Comment:
I'm not a fan of using a macro that's sometimes totally empty and hiding the condition when it becomes a no-op way outside the only line in the file that actually uses `retroshift_swap_times()`.

Reply:
OK. This #if isn't as bad for readability. So it's probably clearer to just leave it there.